src/main/java/io/aiven/kafka/auth/VerdictCache.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Aiven Oy https://aiven.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.auth;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.aiven.kafka.auth.json.AivenAcl;
+import io.aiven.kafka.auth.json.reader.AclJsonReader;
+import io.aiven.kafka.auth.json.reader.JsonReaderException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class VerdictCache {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VerdictCache.class);
+    private List<AivenAcl> aclEntries;
+    private final Map<String, Boolean> cache = new ConcurrentHashMap<>();
+    private final AclJsonReader jsonReader;
```

Comment:
Probably no need to store the reference to `AclJsonReader`, seems more straightforward to pass `aclEntries` to the constructor from `create`. It'll also make `aclEntries` final and solve potential (?) problems with safe publication.